### PR TITLE
Update footer

### LINKED ISSUE #6104: moz.org Footer redesign for Nav/IA updates
Updating the footer is required in order to launch the Nav test

[Nav Outline Copy Doc](https://docs.google.com/document/d/1g0N8oy0WSOY218LJjDIslBbVOmNaBY4Agnfo-cvp6wA/edit?usp=sharing)

### PR DESCRIPTION
**Pull request by stephaniehobson**
## Description
- Updates footer and language switcher markup to match Protocol 4.0.0 for all systems
- Adds [new IA](https://docs.google.com/document/d/1g0N8oy0WSOY218LJjDIslBbVOmNaBY4Agnfo-cvp6wA/edit#heading=h.vvqh5av6uul) to English pages
- Adds new `Supports` and `Utils` files to bundles.

## Issue / Bugzilla link
Fix #6104 

## Testing
- [x] Protocol, Pebbles, and Sandstone pages
  - [x] RTL
  - [x] mobile, tablet, desktop, wide desktop
  - [x] language switcher still works
- [x] Header nav still works? Did I break anything global when we moved stuff into the `Utils` and `Supports` files?

### CODE REVIEW

**Comment by alexgibson** [comment]
I published Protocol 4.0.0 so it's ready to pull into bedrock as a dependency https://www.npmjs.com/package/@mozilla-protocol/core

**Comment by stephaniehobson** [comment]
Down to just a test failure for `test_change_language` progress!

**Comment by stephaniehobson** [comment]
⚡️ Passing ⚡️

**Comment by stephaniehobson** [comment]
P1 because it blocks work to the privacy page. Adjust based on websites team priorities as necessary.

File: bedrock/base/templates/includes/site-footer.html [hunk 1]
```diff
@@ -2,87 +2,8 @@
...
-          </p>
-        </div>
-        <div class="lang-switcher">
-          {% include 'includes/lang_switcher.html' %}
-        </div>
-    </nav>
-  </div>
-</footer>
+{% if LANG.startswith('en-') %}
+   {% include 'includes/protocol/site-footer-en.html' %}
```
Comment (alexgibson) [inline bedrock/base/templates/includes/site-footer.html]: Nit: indentation

File: media/css/pebbles/components/_protocol-footer.scss [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        @include font-size(16px);
+    }
+
+    .mzp-c-footer-secondary {
+        @include font-size(12px);
+    }
+}
+
+
+// bug fix, remove after backported to Protocol
```
Comment (alexgibson) [inline media/css/pebbles/components/_protocol-footer.scss]: Nice catch, shall we file an protocol issue to make sure we don't forget this?

File: media/js/base/mozilla-utils.js [hunk 3]
```diff
@@ -43,21 +43,6 @@ if (typeof Mozilla === 'undefined') {
         return '/' + newLang + '/' + urlpath + location.search;
     };
 
-    // language switcher
-    Utils.initLangSwitcher = function() {
```
Comment (alexgibson) [inline media/js/base/mozilla-utils.js]: There's still a piece of code calling this function. In the console I see:

```javascript
TypeError: utils.initLangSwitcher is not a function
```

File: media/static-bundles.json [hunk 4]
```diff
@@ -311,7 +312,8 @@
     {
       "files": [
         "css/pebbles/global.scss",
-        "css/pebbles/protocol-nav.scss"
+        "css/pebbles/protocol-nav.scss",
+        "css/pebbles/protocol-footer.scss"
```
Comment (alexgibson) [inline media/static-bundles.json]: Can we now remove the `components/footer` import from `global.scss` if all locales are getting the protocol footer? :)

Comment (alexgibson) [reply media/static-bundles.json]: We can likely also remove all the `#colophon {}` styles from Sandstone's monolithic stylesheet?

File: media/js/base/mozilla-utils.js (hunk 3 above)
Comment (alexgibson) [reply media/js/base/mozilla-utils.js]: Can we also remove `Utils.switchPathLanguage` above? (plus the unit tests)

**Comment by alexgibson** [comment]
I see the Protocol 4.0 update also includes a change to our tokens https://github.com/mozilla/protocol-tokens/pull/28/files

I can't see anywhere this will likely impact bedrock, but a second pair of eyes would be good?

File: bedrock/base/templates/base-protocol.html [hunk 5]
```diff
@@ -9,7 +9,7 @@
 {% endblock %}
 
 {% block site_footer %}
-  {% include 'includes/protocol/site-footer.html' %}
+  {% include 'includes/site-footer.html' with context %}
```
[the remark this answers is not in the document]
Comment (stephaniehobson) [reply bedrock/base/templates/base-protocol.html]: ✂️

File: bedrock/base/templates/includes/protocol/site-footer-en.html [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+            <li><a href="https://careers.mozilla.org" data-link-type="footer" data-link-name="Careers">{{ _('Careers') }}</a>
+          </ul>
+        </section>
+        <section class="mzp-c-footer-section">
+          <h5 class="mzp-c-footer-heading">
+            {{ _('Developers') }}
+          </h5>
+          <ul class="mzp-c-footer-list">
+            <li><a href="{{ url('firefox.channel.desktop') }}" data-link-type="footer" data-link-name="Beta, Nightly, Developer Edition">{{ _('Test New Features') }}</a></li>
+            <li><a href="https://developer.mozilla.org" data-link-type="footer" data-link-name="MDN">{{ _('Documentation (MDN)') }}</a></li>
```
[the remark this answers is not in the document]
Comment (stephaniehobson) [reply bedrock/base/templates/includes/protocol/site-footer-en.html]: ✅

File: bedrock/base/templates/includes/protocol/site-footer-en.html [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+            <li><a href="{{ press_blog_url() }}" data-link-type="footer" data-link-name="Press Center">{{ _('Press Center') }}</a>
+            <li><a href="https://careers.mozilla.org" data-link-type="footer" data-link-name="Careers">{{ _('Careers') }}</a>
+          </ul>
+        </section>
+        <section class="mzp-c-footer-section">
+          <h5 class="mzp-c-footer-heading">
+            {{ _('Developers') }}
+          </h5>
+          <ul class="mzp-c-footer-list">
+            <li><a href="{{ url('firefox.channel.desktop') }}" data-link-type="footer" data-link-name="Beta, Nightly, Developer Edition">{{ _('Test New Features') }}</a></li>
```
[the remark this answers is not in the document]
Comment (stephaniehobson) [reply bedrock/base/templates/includes/protocol/site-footer-en.html]: ✅ 

And I went through and added a few more `data-link-name` values.

File: bedrock/base/templates/includes/protocol/site-footer-en.html [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+          {% trans url='https://foundation.mozilla.org', link_type='footer', link_name='Mozilla Foundation' %}
+            Visit Mozilla Corporation’s not-for-profit parent, the <a href="{{ url }}" data-link-type="{{ link_type }}" data-link-name="{{ link_name }}">Mozilla Foundation</a>.
+          {% endtrans %}<br>
+          {%- trans url=url('foundation.licensing.website-content') -%}
+              Portions of this content are ©1998–{{ current_year }} by individual mozilla.org
+              contributors. Content available under a <a rel="license" href="{{ url }}">Creative Commons license</a>.
+          {%- endtrans -%}
+        </p>
+        <ul class="mzp-c-footer-terms">
+          {% if LANG.startswith('en-') %}
```
[the remark this answers is not in the document]
Comment (stephaniehobson) [reply bedrock/base/templates/includes/protocol/site-footer-en.html]: I think so? Not sure why were were only showing one of the two links to non English users. There are localizations of both available.

File: bedrock/base/templates/includes/protocol/site-footer.html [hunk 9]
```diff
@@ -2,85 +2,79 @@
...
+            {{ _('Mozilla') }}
+          </h5>
+          <ul class="mzp-c-footer-list">
+            <li><a href="{{ url('mozorg.about') }}" data-link-type="footer" data-link-name="About">{{ _('About') }}</a></li>
+            <li><a href="https://blog.mozilla.org/" data-link-type="footer" data-link-name="Blog">{{ _('Blog') }}</a></li>
+            <li><a href="{{ url('mozorg.contact.contact-landing') }}" data-link-type="footer" data-link-name="Contact Us">{{ _('Contact Us') }}</a></li>
+            <li><a href="{{ press_blog_url() }}" data-link-type="footer" data-link-name="Press Center">{{ _('Press Center') }}</a>
+            <li><a href="{{ donate_url('footer') }}" class="donate" data-link-type="footer" data-link-name="Donate">{{ _('Donate') }}</a></li>
+            <li><a href="https://wiki.mozilla.org/Webdev/GetInvolved/mozilla.org" data-link-type="footer" data-link-name="Contribute to this site">{{ _('Contribute to this site') }}</a></li>
+            {# TODO remove the "and" clause after tranlsation #}
```
[the remark this answers is not in the document]
Comment (stephaniehobson) [reply bedrock/base/templates/includes/protocol/site-footer.html]: ✂️  It's not in the new footer, and now it's not in the old one either.

File: bedrock/base/templates/includes/protocol/site-footer.html [hunk 10]
```diff
@@ -2,85 +2,79 @@
...
-          </p>
-        </div>
-        {% include 'includes/protocol/lang-switcher.html' %}
+        {%- trans url=url('foundation.licensing.website-content') -%}
+          Portions of this content are ©1998–{{ current_year }} by individual mozilla.org
+          contributors. Content available under a <a rel="license" href="{{ url }}">Creative Commons license</a>.
+        {%- endtrans -%}
+
+        <ul class="mzp-c-footer-terms">
+          {% if LANG.startswith('en-') %}
```
[the remark this answers is not in the document]
Comment (stephaniehobson) [reply bedrock/base/templates/includes/protocol/site-footer.html]: ✂️

File: bedrock/base/templates/includes/site-footer.html (hunk 1 above)
Comment (stephaniehobson) [reply bedrock/base/templates/includes/site-footer.html]: ✅

File: media/css/pebbles/components/_protocol-footer.scss (hunk 2 above)
Comment (stephaniehobson) [reply media/css/pebbles/components/_protocol-footer.scss]: "we" 😝

https://github.com/mozilla/protocol/issues/317

File: media/static-bundles.json (hunk 4 above)
Comment (stephaniehobson) [reply media/static-bundles.json]: Yes. Yes we can!

Comment (stephaniehobson) [reply media/static-bundles.json]: ✂️

**Comment by stephaniehobson** [comment]
Looks like the token change breaks /firefox/new/?xv=edge at *precisely* 480px.

File: media/js/base/mozilla-utils.js (hunk 3 above)
Comment (stephaniehobson) [reply media/js/base/mozilla-utils.js]: I can't find the string `initLangSwitcher` in my code base.

✂️ `switchPathLanguage`

**Comment by stephaniehobson** [comment]
Updated!

**Comment by alexgibson** [comment]
Think you may have forgotten to push to the remote branch? I don't see any changes yet

**Comment by stephaniehobson** [comment]
Yup. Pushed to stephaniehobson/master and not mozilla/master 🤦‍♀️

File: bedrock/base/templates/includes/protocol/site-footer-en.html [hunk 11]
```diff
@@ -24,8 +24,8 @@ <h5 class="mzp-c-footer-heading">
             {{ _('Developers') }}
           </h5>
           <ul class="mzp-c-footer-list">
-            <li><a href="{{ url('firefox.channel.desktop') }}" data-link-type="footer" data-link-name="Beta, Nightly, Developer Edition">{{ _('Test New Features') }}</a></li>
-            <li><a href="https://developer.mozilla.org" data-link-type="footer" data-link-name="MDN">{{ _('Documentation (MDN)') }}</a></li>
+            <li><a href="{{ url('firefox.channel.desktop') }}" data-link-type="footer" data-link-name="Test New Features">{{ _('Test New Features') }}</a></li>
+            <li><a href="https://developer.mozilla.org" data-link-type="footer" data-link-name="MDN">{{ _('MDN Web Docs') }}</a></li>
```
Comment (alexgibson) [inline bedrock/base/templates/includes/protocol/site-footer-en.html]: `data-link-name="MDN Web Docs"` ?

**Comment by alexgibson** [comment]
On pebbles pages such as /firefox/ the footer headings aren't bold, making them hard to distinguish from the links.

screenshot_2019-02-20 the new fast browser for mac pc and linux firefox

File: bedrock/base/templates/includes/protocol/site-footer-en.html [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+      </div>
+      <div class="mzp-c-footer-sections">
+        <section class="mzp-c-footer-section">
+          <h5 class="mzp-c-footer-heading">
+            {{ _('Company') }}
+          </h5>
+          <ul class="mzp-c-footer-list">
+            <li><a href="{{ url('mozorg.about') }}" data-link-type="footer" data-link-name="About">{{ _('About') }}</a></li>
+            <li><a href="{{ press_blog_url() }}" data-link-type="footer" data-link-name="Press Center">{{ _('Press Center') }}</a>
+            <li><a href="https://careers.mozilla.org" data-link-type="footer" data-link-name="Careers">{{ _('Careers') }}</a>
```
[the remark this answers is not in the document]
Comment (alexgibson) [reply bedrock/base/templates/includes/protocol/site-footer-en.html]: Ooh, should we still do this?

[the remark this answers is not in the document]
Comment (stephaniehobson) [reply bedrock/base/templates/includes/protocol/site-footer-en.html]: Added them for all mozilla.org properties.

I suppose I should put this up on a demo and get an analytics review.

File: bedrock/base/templates/includes/protocol/site-footer-en.html (hunk 11 above)
Comment (stephaniehobson) [reply bedrock/base/templates/includes/protocol/site-footer-en.html]: ✅

**Comment by stephaniehobson** [comment]
Updated and looking for someone to do an analytics review.

**Comment by stephaniehobson** [comment]
Oh, and yes to localization 💯 

Can we merge and then talk to the l10n team? Or should I do that first?

**Comment by ejregithub** [comment]
@peiying2 - I tagged this one for l10n so we can get that ball rolling.  Let us know of any questions about the content/strings, please.

**Comment by alexgibson** [comment]
@stephaniehobson looks like this needs a rebase. In the interests of moving forward with updating bedrock to 4.0, would moving forward with what we have now be an option from an analytics point of view? We could always update any UTM params in a post launch fix perhaps - thoughts?

**Comment by stephaniehobson** [comment]
I like the way you think.

**Comment by stephaniehobson** [comment]
Oops, that was supposed to be a squash and merge not a rebase and merge. I guess there's not anything I can do about that now?

**Comment by alexgibson** [comment]
Something is not quite right here:

[package.json](https://github.com/mozilla/bedrock/blob/master/package.json#L7) list protocol at 4.0.0 as expected, but the [yarn lockfile](https://github.com/mozilla/bedrock/blob/master/yarn.lock#L22) still says 3.1.0, suggesting Yarn wasn't used to update the dependency. PR incoming.

**Comment by stephaniehobson** [comment]
@rraue New footer is merged! It will probably go live tomorrow.

**Pull request merged**